src/frontend/screens/Settings/MapSettings/BackgroundMapInfo.tsx

```diff
@@ -0,0 +1,184 @@
+import { BottomSheetMethods } from "@gorhom/bottom-sheet/lib/typescript/types";
+import MapboxGL from "@react-native-mapbox-gl/maps";
+import * as React from "react";
+import { defineMessages, useIntl } from "react-intl";
+import { ScrollView, StyleSheet, Text, View } from "react-native";
+
+import api from "../../../api";
+import { MAPEO_BLUE, MEDIUM_GREY, WHITE } from "../../../lib/styles";
+import Button from "../../../sharedComponents/Button";
+import Loading from "../../../sharedComponents/Loading";
+import { NativeRootNavigationProps } from "../../../sharedTypes";
+import { DeleteMapBottomSheet } from "./DeleteMapBottomSheet";
+import { useMapStyle } from "../../../hooks/useMapStyle";
+import { convertBytesToMb, DEFAULT_MAP_ID } from "./BackgroundMaps";
+import { DeleteIcon } from "../../../sharedComponents/icons";
+
+const m = defineMessages({
+  removeMap: {
+    id: "screens.Settings.MapSettings.removeMap",
+    defaultMessage: "Remove Map",
+  },
+  cancel: {
+    id: "screens.Settings.MapSettings.cancel",
+    defaultMessage: "Cancel",
+  },
+  subtitle: {
+    id: "screens.Settings.MapSettings.subtitle",
+    defaultMessage:
+      "This map and offline areas attached to it will be deleted. This cannot be undone",
+  },
+  mb: {
+    id: "screens.Settings.MapSettings.mb",
+    defaultMessage: "MB",
+    description: "abbreviation for megabyte",
+  },
+  zoomLevel: {
+    id: "screens.Settings.MapSettings.zoomLevel",
+    defaultMessage: "Zoom Level",
+  },
+  description: {
+    id: "screens.Settings.MapSettings.description",
+    defaultMessage: "Description",
+  },
+  deleteMap: {
+    id: "screens.Settings.MapSettings.deleteMap",
+    defaultMessage: "Delete Map",
+  },
+  useMap: {
+    id: "screens.Settings.MapSettings.useMap",
+    defaultMessage: "Use Map",
+  },
+});
+
+export const BackgroundMapInfo = ({
+  route,
+  navigation,
+}: NativeRootNavigationProps<"BackgroundMapInfo">) => {
+  const { formatMessage: t } = useIntl();
+  const { bytesStored, id, styleUrl, name } = route.params;
+  const [zoomAndDescription, setZoomAndDescription] = React.useState<
+    { zoom?: number; description?: string } | "loading"
+  >("loading");
+
+  const sheetRef = React.useRef<BottomSheetMethods>(null);
+
+  const { setStyleId } = useMapStyle();
+
+  function setStyleAndNavigateHome() {
+    setStyleId(id);
+    navigation.navigate("Home", { screen: "Map" });
+  }
+
+  React.useEffect(() => {
+    api.maps
+      .getStyle(id)
+      .then(style => {
+        // @ts-ignore
+        return api.maps.getTileset(style.sources["raster-source"].url);
+      })
+      .then(obj => {
+        setZoomAndDescription({
+          zoom: obj.maxzoom.toString() || undefined,
```

Comment:
nit: does the zoom need to be a string in any case? where you're rendering it in the JSX doesn't require it to be so

Reply:
Curious why typescipt is not picking this up as an error, given that the typing of `zoom` is `number`

Reply:
maybe due to the `@ts-ignore` you have above?

Reply:
oh no, its actually because obj.maxzoom is of type `any`. Aka there is no associated typing

Reply:
it's type any because you're ignoring the return value in the promise in the line above. if you remove that and update the return type for the api call, then the `obj` type will be properly annotated

Reply:
right, but actually he reason i am doing that is because if this error.
<img width="811" alt="image" src="https://user-images.githubusercontent.com/67773827/198374376-4b0f9487-5ca2-4da1-9523-3507f290871c.png">

Reply:
With` style.sources`, i am doing something wrong (I havent updated this endpoint yet, but the error is the `.url`). Any idea what is going on here

Reply:
ah yeah you need to check for the existence of the property first and then access it. It's kind of a TS-specific thing but i think something along the lines of this should work :


```ts
const source = style.sources["raster-source"]
if ('url' in source) {
  // Do stuff with `source.url`
}
```

Reply:
That error message is misleading...didn't realize that this was a simple optional/undefined type error

Reply:
yeah it could be better but it's slightly different than what you're used to seeing because of how discriminate unions work, so a different kind of narrowing is needed

looks like this piece explains it well enough to provide context: https://css-tricks.com/typescript-discriminated-unions/

Reply:
interesting way of narrowing that is done in that article. Perhaps something we can adopt for some of our complicated types